### DISCUSSION TITLE
Tree-sitter on terminal buffers

### DISCUSSION THREAD

**Discussion by bqv** [neovim/neovim#31304] [General]
Can I get a second opinion on https://github.com/nvim-treesitter/nvim-treesitter/issues/7388

**Comment by clason** [comment]
Still no :)

**Reply by bqv** [reply]
Any detail on why exactly? As I said, it half-works already

**Reply by clason** [reply]
"Half" is a massive overstatement; the most important part is syncing the buffer with treesitter. And there is no infrastructure for this, since the terminal buffer is not managed by Neovim itself but by the shell (i.e., we just draw what the application tells the terminal to draw using standard ANSI sequences, there is no separate "highlighting" loop like for actual buffers).
